source/_posts/2017-02-07-falsehoods-programmers-believe-about-language.markdown

```diff
@@ -0,0 +1,189 @@
+---
+layout: post
+title: "Falsehoods Programmers Believe About Language"
+date: 2017-02-07 22:17:11 -0600
+comments: true
+categories: 
+---
+
+As is now some kind of traditoin akin to "X Considered Harmful" posts of old,
+here I am, a software developer, writing a post trying to lay bare some of the
+[wrong ideas many developers hold about a broad topic](https://github.com/jameslk/awesome-falsehoods).
+In keeping with my degree and a personal area of interest, let's look at some of
+the things (Anglophonic) developers often believe about language that do not
+_quite_ track to reality.
+
+
+## Translating my application into other languages isn't important.
+
+It's true that many people in the world who speak more than one language choose
+to learn English and that many of those people speak "better" English than many
+native English speakers. (I put "better" in quotes because that idea is pretty
+judgemental against certain dialects of English that often have racist and/or
+classist components).
+
+However, there's a big difference between being _able_ to use an application and
+being _comfortable_ using an application. If two competing applications exist
+and one is in a user's native language, they are very likely to pick that one
+even if it doesn't have all the bells and whistles.
+
+
+## Translating a single word of English is simple.
+
+Think of all the places in software where just a single word appears. If a
+translator is only presented with a YAML file of strings, how are they supposed
+to know whether it's "Back" as in "go back to where you were before" or as in
+"not the front"? Whether it's "Post" as in "post my comment, now" or as in
+"mail"? This can lead to some very confusing experiences for non-English users.
+
+Also, if we consider just verbs, often tenses collide in English. "Eat" appears
+in all of: "I will eat this" (future), "I eat this (often)" (habitual present),
+"Did you eat this" (past) and "I was going to eat that" (conditional perfect, I
+think. Who even names these things?!?).
+
+If we consider just _nouns_, some lanuages have what's called a case system
+which is used instead of word order and/or pronouns to denote a noun's
+relationship to the rest of the words in a sentence. So "gift" could have a
+different shape in translations of these sentences:
+
+* The gift is ready.
+* I gave the gift to you.
+* I threw a ball at the gift.
+* The gift's wrapping is pretty.
+* This is my gift.
```

Comment:
If you had the spanish/japanese/whatever equivalents of each of these sentences and gift was spelled differently in each of them the impact of this would be even larger